be/src/olap/field.h

```diff
@@ -278,7 +278,7 @@ class Field {
 
     FieldType type() const { return _type_info->type(); }
     FieldAggregationMethod aggregation() const { return _agg_info->agg_method(); }
-    const TypeInfo* type_info() const { return _type_info; }
+    const TypeInfo* type_info() const { return _type_info.get(); }
```

Comment:
Is there a situation where the type info pointer is held, but the smart pointer has been released？

Reply:
I agree that it is a potential risk which we should avoid.